src/applications/check-in/day-of/pages/LoadingPage/index.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { useDispatch, useSelector, batch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { URLS } from '../../../utils/navigation/day-of';
+import { useFormRouting } from '../../../hooks/useFormRouting';
+import {
+  receivedEmergencyContact,
+  receivedDemographicsData,
+  receivedNextOfKinData,
+  receivedMultipleAppointmentDetails,
+  triggerRefresh,
+  updateFormAction,
+} from '../../../actions/day-of';
+import { useSessionStorage } from '../../../hooks/useSessionStorage';
+import { makeSelectCheckInData } from '../../hooks/selectors';
+import { api } from '../../../api';
+
+const LoadingPage = props => {
+  const {
+    isSessionLoading,
+    router,
+    isEmergencyContactEnabled,
+    checkInExperienceUpdateInformationPageEnabled,
+  } = props;
+  const { goToErrorPage, goToNextPage } = useFormRouting(router, URLS);
+  const selectCheckInData = useMemo(makeSelectCheckInData, []);
+  const checkInData = useSelector(selectCheckInData);
+  const { getCurrentToken } = useSessionStorage(false);
+  const { context, appointments } = checkInData;
+  const dispatch = useDispatch();
+  const setSessionData = useCallback(
+    (payload, token) => {
+      batch(() => {
+        const {
+          appointments: appts,
+          demographics: demo,
+          patientDemographicsStatus,
+        } = payload;
+        dispatch(triggerRefresh(false));
+        dispatch(receivedMultipleAppointmentDetails(appts, token));
+        dispatch(
+          updateFormAction({
+            patientDemographicsStatus,
+            isEmergencyContactEnabled,
+            checkInExperienceUpdateInformationPageEnabled,
+          }),
+        );
+        if (typeof demo !== 'undefined') {
+          dispatch(receivedDemographicsData(demo));
+          if ('nextOfKin1' in demo) {
+            dispatch(receivedNextOfKinData(demo.nextOfKin1));
+          }
+          if ('emergencyContact' in demo) {
+            dispatch(receivedEmergencyContact(demo.emergencyContact));
+          }
+        }
+      });
+    },
+    [
+      dispatch,
+      isEmergencyContactEnabled,
+      checkInExperienceUpdateInformationPageEnabled,
+    ],
+  );
+
+  useEffect(
+    () => {
+      let isCancelled = false;
+      const session = getCurrentToken(window);
+      if (!context || !session) {
```

Comment:
Do we need the session check if we are still using the `withSession`?

Reply:
oh good question, I'll look into that. I think I just tried to copy/paste as much of our existing flow as I could.

Reply:
removed session check